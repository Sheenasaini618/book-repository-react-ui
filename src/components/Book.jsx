import * as React from 'react';
import Card from '@mui/material/Card';
import CardHeader from '@mui/material/CardHeader';
import CardMedia from '@mui/material/CardMedia';
import CardContent from '@mui/material/CardContent';
import CardActions from '@mui/material/CardActions';
import Avatar from '@mui/material/Avatar';
import Typography from '@mui/material/Typography';
import { red } from '@mui/material/colors';
import ManualEditBook from './ManualEditBook';
import DeleteBook from './DeleteBook';


export default function Book(props) {

  var book = props.details
  console.log(book)
  return (
    <Card sx={{ maxWidth: 450, maxHeight: 500 }} style={{ marginTop: "10px" }}>
      <CardHeader
        avatar={
          <Avatar sx={{ bgcolor: red[500] }} aria-label="recipe">
            {book.title.charAt(0).toUpperCase()}
          </Avatar>
        }
        title={book.title}
        subheader={book.authors[0]}
      />
      <CardMedia
        component="img"
        sx={{ width: 151 }}
        height="194"
        image={book.imageLinks.smallThumbnail}
        alt="Paella dish"
      />
      <CardContent>
        <Typography variant="h6" color="text.secondary" align='justify' style={{ fontWeight: 600 }}>
          Price: â‚¹{book.price}.00
        </Typography>
        <Typography variant="h6" color="text.secondary" align='justify' style={{ fontWeight: 600 }}>
          Copies Available: {book.quantity}
        </Typography>
      </CardContent>
      <CardActions disableSpacing>
        <ManualEditBook details={book} />
        <DeleteBook details={book.id} />
      </CardActions>
    </Card>
  );
}
