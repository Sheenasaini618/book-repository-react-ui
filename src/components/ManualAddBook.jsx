import * as React from "react";
import Button from "@mui/material/Button";
import TextField from "@mui/material/TextField";
import Dialog from "@mui/material/Dialog";
import DialogActions from "@mui/material/DialogActions";
import DialogContent from "@mui/material/DialogContent";
import DialogTitle from "@mui/material/DialogTitle";
import { addBook } from "../util/api/bookApi";

export default function ManualAddBook() {
  const [open, setOpen] = React.useState(false);
  const [formData, setFormData] = React.useState({
    title: "",
    thumbnail: "",
    authors: "",
    description: "",
    price: "",
    quantity: "",
  });

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleAddBook = () => {
    addBook(formData).then((res) => {
      console.log(res);
      handleClose();
      window.history.go("http://localhost:3000");
    });
  };

  const handleChange = (e) => {
    setFormData({
      ...formData,
      [e.target.id]: e.target.value,
    });
  };

  return (
    <center>
      <div style={{ margin: "10px" }}>
        <Button onClick={handleClickOpen} variant="contained" color="info">
          Add Book using Form
        </Button>
        <Dialog open={open} onClose={handleClose}>
          <DialogTitle> Add Book using Form</DialogTitle>
          <DialogContent>
            <TextField
              autoFocus
              margin="dense"
              id="title"
              label="Title"
              fullWidth
              variant="standard"
              value={formData.title}
              onChange={handleChange}
            />
            <TextField
              autoFocus
              margin="dense"
              id="thumbnail"
              label="Thumbnail Link"
              fullWidth
              variant="standard"
              value={formData.thumbnail}
              onChange={handleChange}
            />
            <TextField
              autoFocus
              margin="dense"
              id="authors"
              label="Authors"
              fullWidth
              variant="standard"
              value={formData.authors}
              onChange={handleChange}
            />
            <TextField
              autoFocus
              margin="dense"
              id="price"
              label="Price"
              type="number"
              fullWidth
              variant="standard"
              value={formData.price}
              onChange={handleChange}
            />
            <TextField
              autoFocus
              margin="dense"
              id="quantity"
              label="Quantity"
              fullWidth
              type="number"
              variant="standard"
              value={formData.quantity}
              onChange={handleChange}
            />
            <TextField
              autoFocus
              margin="dense"
              id="description"
              label="Description"
              fullWidth
              variant="standard"
              value={formData.description}
              onChange={handleChange}
            />
          </DialogContent>
          <DialogActions>
            <Button onClick={handleClose}>Cancel</Button>
            <Button onClick={handleAddBook}>Add Book</Button>
          </DialogActions>
        </Dialog>
      </div>
    </center>
  );
}
