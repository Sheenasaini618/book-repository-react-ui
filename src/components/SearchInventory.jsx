import * as React from 'react';
import Paper from '@mui/material/Paper';
import InputBase from '@mui/material/InputBase';
import Divider from '@mui/material/Divider';
import { retrieveBooksBasisTitle } from '../util/api/bookApi';
import BookCatalogue from './BookCatalogue';

export default function SearchInventory() {

    const [searchBookInventory, setsearchBookInventory] = React.useState("");
    const [data, setData] = React.useState([]);

    const handleChange = (e) => {
        setsearchBookInventory(
            e.target.value);
        console.log(e.target.value)
    };

    //componentDidMount and componentDidUpdate

    React.useEffect(() => {
        //Promise<Response>
        //Response
        console.log(searchBookInventory)
        retrieveBooksBasisTitle(searchBookInventory)
            .then((res) => {
                return res.json();
            })
            .then((result) => {
                setData(result);
            });
    }, [searchBookInventory]);

    return (
        <>
            <Paper
                component="form"
                sx={{ p: '2px 4px', display: 'flex', alignItems: 'center', width: 400 }}>
                <InputBase className='placeholder'
                    sx={{ ml: 1, flex: 1 }}
                    placeholder="Search Inventory book by title or author"
                    onChange={handleChange}
                />
                <Divider sx={{ height: 28, m: 0.5 }} orientation="vertical" />
            </Paper>
            <BookCatalogue details={data} />
        </>
    );
}
