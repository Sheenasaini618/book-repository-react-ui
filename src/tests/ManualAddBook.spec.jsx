import React from 'react'
import Book from '../components/Book'
import '@testing-library/jest-dom'
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme'
import { configure } from 'enzyme';
import SearchInventory from '../components/SearchInventory';
import ManualAddBook from '../components/ManualAddBook';


configure({ adapter: new Adapter() });

let component

beforeEach(() => {
    component = shallow(<ManualAddBook />);
});

global.fetch = jest.fn().mockImplementation(() =>
    Promise.resolve({
        ok: true,
        status: 200,
    }),
);

describe('<SearchInventory />', () => {

    let wrapper;

    if (typeof window === 'undefined') {
        global.window = {}
    }

    it('should call keyPress onChange of fullName with error message', () => {

        wrapper = shallow(<ManualAddBook />);
        const event = {};
        expect(wrapper.find('.clickButton').length).toBe(1);
    })

    it('should match snapshot', () => {
        const element = component.debug()
        expect(element).toMatchSnapshot();
    });

    it('should validate elements in manual add book component', () => {

        wrapper = shallow(<ManualAddBook />);
        const event = {};
        expect(wrapper.find('.clickButton').length).toBe(1);
        wrapper.find('.closeButton').at(0).simulate('click', event);
        wrapper.find('.addBookButton').at(0).simulate('click', event);
    })

})
