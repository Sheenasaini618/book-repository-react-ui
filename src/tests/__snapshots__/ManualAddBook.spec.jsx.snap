// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInventory /> should render book  1`] = `
"<center>
  <div style={{...}}>
    <ForwardRef(Button) className=\\"clickButton\\" onClick={[Function: handleClickOpen]} variant=\\"contained\\" color=\\"info\\">
      Add Book using Form
    </ForwardRef(Button)>
    <ForwardRef(Dialog) open={false} onClose={[Function: handleClose]}>
      <ForwardRef(DialogTitle)>
         Add Book using Form
      </ForwardRef(DialogTitle)>
      <ForwardRef(DialogContent)>
        <ForwardRef(TextField) autoFocus={true} margin=\\"dense\\" id=\\"title\\" label=\\"Title\\" fullWidth={true} variant=\\"standard\\" value=\\"\\" onChange={[Function: handleChange]} />
        <ForwardRef(TextField) autoFocus={true} margin=\\"dense\\" id=\\"thumbnail\\" label=\\"Thumbnail Link\\" fullWidth={true} variant=\\"standard\\" value=\\"\\" onChange={[Function: handleChange]} />
        <ForwardRef(TextField) autoFocus={true} margin=\\"dense\\" id=\\"authors\\" label=\\"Authors\\" fullWidth={true} variant=\\"standard\\" value=\\"\\" onChange={[Function: handleChange]} />
        <ForwardRef(TextField) autoFocus={true} margin=\\"dense\\" id=\\"price\\" label=\\"Price\\" type=\\"number\\" fullWidth={true} variant=\\"standard\\" value=\\"\\" onChange={[Function: handleChange]} />
        <ForwardRef(TextField) autoFocus={true} margin=\\"dense\\" id=\\"quantity\\" label=\\"Quantity\\" fullWidth={true} type=\\"number\\" variant=\\"standard\\" value=\\"\\" onChange={[Function: handleChange]} />
        <ForwardRef(TextField) autoFocus={true} margin=\\"dense\\" id=\\"description\\" label=\\"Description\\" fullWidth={true} variant=\\"standard\\" value=\\"\\" onChange={[Function: handleChange]} />
      </ForwardRef(DialogContent)>
      <ForwardRef(DialogActions)>
        <ForwardRef(Button) onClick={[Function: handleClose]}>
          Cancel
        </ForwardRef(Button)>
        <ForwardRef(Button) onClick={[Function: handleAddBook]}>
          Add Book
        </ForwardRef(Button)>
      </ForwardRef(DialogActions)>
    </ForwardRef(Dialog)>
  </div>
</center>"
`;
